### PR TITLE
Gazoline Fuel Cell cbm

### PR DESCRIPTION
**Pull request by Fris0uman**
#### Summary
```SUMMARY: Content "Add Gazoline Fuel Cell CBM"```

#### Purpose of change
Gazoline Fuel Cell CBM

#### Describe the solution
Add Gazoline Fuel Cell CBM: can hold 500 ml of gazolinel, consume 1 ml/s and produce about 8 power/s
It adds 5 torso encumbrance

#### Additional context
I'll try to add the code support for it to actually produce heat in another PR.

### CODE REVIEW

File: data/json/bionics.json [hunk 1]
```diff
@@ -1110,6 +1110,18 @@
     "occupied_bodyparts": [ [ "ARM_R", 1 ], [ "HAND_R", 2 ] ],
     "act_cost": 50
   },
+  {
+    "id": "bio_diesel_generator",
+    "type": "bionic",
+    "name": "Diesel Generator CBM",
+    "description": "A small diesel generator inserted into you're abdominal cavity, a set of metal bracings spread its weight against your bone structure to minimize discomfort.  A diffuse network of bio-plastic bladders has been meshed with your circulator system and serves as diesel tank.",
```
Comment (anothersimulacrum) [inline data/json/bionics.json]:
```suggestion
    "description": "A small diesel generator inserted into your abdominal cavity, with a set of metal bracings to spread its weight against your bone structure to minimize discomfort.  A diffuse network of bio-plastic bladders has been meshed with your circulatory system and serves as diesel tank.",
```

**Comment by kevingranade** [comment]
Ok, so a compact gasoline fuel cell is a thing https://www.technologyreview.com/s/426252/gasoline-fuel-cell-would-boost-electric-car-range/

Efficiency wise, a quick survey says they may be about 50% efficient, so convert half of the 45 MJ/Kg available in gasoline to electricity.
That's like.... 15,000 charges per mL?
Can any chemistry peeps double check me? That sounds crazy.

**Comment by justamblingalong** [comment]
Is there a standard definition of what a CBM charge is in real-world terms? Happy to do some calcs (physics background here) but not sure where to start.

It definitely seems like an attempt to enforce physics-based numbers on the CBM system would result in either a) most CBMs being free or b) most CBMs being unusably expensive, depending on where the 'unit' was set; some CBMs, like the Chain Lightning Generator or the Shockwave Generator, would require multiple orders of magnitude more power than, say, Night Vision (assuming the primary power draw is cooling the system) or certainly the Cranial Flashlight. 

A more realistic system would also definitely run hard into what you're finding above, which is that some sources of power (like fission and diesel engines) are going to be totally incomparable to others (like Joint Torsion and Metabolic Interchange). Again, by multiple orders of magnitude.

**Comment by nexusmrsep** [comment]
Also, if the CBM uses extrapolation of that technology in the article, it might overheat it's user, which would be in itself a pretty interesting limiting factor of it's use, if it were to be coded in.

**Comment by jeremyshannon** [comment]
Fuel cells have crazy huge energy output, which is why a lot of R&D is going on to make them viable. It's hard to beat the energy density of liquid fuels.

**Comment by Rail-Runner** [comment]
It'd also have to be noisy as well, I think. Internal combustion engines are never silent, as far as I know.

**Comment by Firestorm01X2** [comment]
Ok. 
Internal Furnace CBM - was removed: 
https://github.com/CleverRaven/Cataclysm-DDA/pull/30228

Because:
>The internal furnace is comically impractical and the bionics system is moving away from such per...

Solar CBM was removed
https://github.com/CleverRaven/Cataclysm-DDA/pull/23317
Because (https://github.com/CleverRaven/Cataclysm-DDA/pull/20696#issuecomment-289637185):
>Scaling via coverage is an option, but I'd prefer to just ditch the solar
panel CBM, it's utter nonsense.

But. Dieasel generator it tottally ok. I mean just imagine.

You are sitting home on friday with cup of tea and your cat and reading book lighting it with your cranial flashlight CBM.

But suddenly you cranial flaslight turned off. You are thinking - "OK, My battery needs recharging". And then you are getting cainster under the table, emeptiing it inside of your body. And then starting cracting diesel generator.

Cat is shocked. Neigbors calling fire team. You staring suffocating with smoke with your cat.

This is not silly at all. Right.

**Comment by Rail-Runner** [comment]
> But. Internal Dieasel generator is totally ok. I mean just imagine.
> 
> You are sitting home on friday with cup of tea and your cat and reading book lighting it with your cranial flashlight CBM.
> 
> But suddenly you cranial flaslight turned off. You are thinking - "OK, TIME TO RECHARGE MY BATTERY!!!".
> And then you are getting cracking fuel cainster under the table, emeptiing it inside of your body. And then you are starting cracking diesel generator.
> 
> Cat is shocked. Neigbors calling fire team. You staring suffocating with smoke with your cat.
> 
> This is not silly at all. And much practical.
> 
> Steampunk on the march.

Well, this wouldn't be something you should be using indoors or with people sleeping nearby, but otherwise it could work. So it's likely to be available mostly for non-civilian uses.

**Comment by nexusmrsep** [comment]
I believe we're talking about a exothermic chemical reaction, not actual combustion with fire smoke etc. Or am I wrong.

**Comment by Firestorm01X2** [comment]
> I believe we're talking about a exothermic chemical reaction, not actual combustion with fire smoke etc. Or am I wrong.

I don't know about what we are talking. But we definitely should be talking about game design, planning and consistency.

**Comment by Firestorm01X2** [comment]
Here is two things:
1) From gameplay point perspective this CBM totally fine and should be added. It adds variesty.
2) From **realism** point erspective it is silly and imparactical. Wha in the right mind wil install diesel generator. with canister inside instead of just connecting to it?

If you choose 1) then all removed CBMs should be added back. If not - then 1) is not applicable.

If you choose 2) then this PR should be closed, becasue it violates game design ideas that was used before to remove few CBMs

So what to choose?

**Comment by jeremyshannon** [comment]
A fuel cell doesn't use combustion, and is quieter and produces less emissions than a combustion engine. It is also more efficient, achieving nearly 60% efficiency given clean, pure air and fuel to work with. They're usually run on hydrogen, but can be made to utilize almost any fuel you could burn.  They do tend to produce a lot of heat, though.
 This is not just a tiny vehicle engine, fuel cells' purpose is basically a battery that you can refill. The metabolic interchange appears to be a fuel cell run on blood glucose.

**Comment by Firestorm01X2** [comment]
> A fuel cell doesn't use combustion, and is quieter and produces less emissions than a combustion engine

Where do you get that? What fuel cell?

> "description": "A small diesel generator inserted into your abdominal cavity, with a set of metal bracings spreading its weight against your bone structure to minimize discomfort.  A diffuse network of bio-plastic bladders has been meshed with your circulatory system and serves as diesel tank.",

**Comment by Rail-Runner** [comment]
Well, now thinking about it, in my opinion it'd make the most sense to have a "portable diesel generator" item, which can efficiently recharge bionic power via existing cable charger CBM, or anything else as needed. There's indeed not much point into building such a generator into a body if it could be carried.

**Comment by Firestorm01X2** [comment]
> Well, now thinking about it, in my opinion it'd make the most sense to have a "portable diesel generator"

Exactly that. Yes.

Just like here:
https://github.com/CleverRaven/Cataclysm-DDA/pull/24027

Or we can just stop playing psuedo relism things and just adding it right as it is.

**Comment by jeremyshannon** [comment]
>Where do you get that?
From Kevin, upthread. If the PR's description says it literally stuffs a generator in your body, that IS silly and should be changed. A fuel cell would be reasonable to use for this which I assume is why Kevin brought it up.

**Comment by Firestorm01X2** [comment]
> 
> 
> > Where do you get that?
> > From Kevin, upthread. If the PR's description says it literally stuffs a generator in your body, that IS silly and should be changed. A fuel cell would be reasonable to use for this which I assume is why Kevin brought it up.

https://www.quora.com/Is-there-a-gasoline-fuel-cell
> One of the big problems is that the fuel cell catalysts are often poisoned by the co-production of carbonyl compounds (not to mention impurities such as sulfur compounds).

Looks like putting it inside body is great way to kill yourself.

**Comment by jeremyshannon** [comment]
What makes you think its designers would make it dump its waste products into your bloodstream? An exhaust port with a tiny bladder that could be emptied would be a saner choice.

**Comment by Firestorm01X2** [comment]
> exhaust port with a tiny bladder that could be emptied would be a saner choice.

It is still posionus and dangerous. Sou I guess user should be very caraefull with
>tiny bladder

And there is still question - why are going to install that kind of thing inside instead of just connecting to external engine?

From gameplay perpective it is totally fine. From realism perspective it is quite stupid idea.

**Comment by jeremyshannon** [comment]
>why are going to install that kind of thing inside instead of just connecting to external engine?

These ARE the people responsible for the various broken/prototype cyborgs and unwilling mutants. The advantages of a built-in massive bionic power generation system may very well outweigh their fears for the safety of whoever has to have it installed so they can fight the Chinese.
 I had a history teacher who was exposed to radiation from a nuclear bomb test as part of a DoD study back in WWII.

**Comment by Firestorm01X2** [comment]
>The advantages of a built-in massive bionic power generation system may very well outweigh their fears for the safety of whoever has to have it installed so they can fight the Chinese.

For crazy expriments - yes. But for civilian use CBM?

**Comment by jeremyshannon** [comment]
Is this an off-the-shelf civilian CBM? I don't think it should be, unless they've got the safety factor worked out well enough to satisfy CDDA Joe Public, although I gather that's a much lower bar than in our universe.

**Comment by Firestorm01X2** [comment]
> Is this an off-the-shelf civilian CBM? I don't think it should be, unless they've got the safety factor worked out well enough to satisfy CDDA Joe Public, although I gather that's a much lower bar than in our universe.

As I said this way we weill end up to teleport positions sumproduct to another dimension via internal teleport which is part of the CBM.

Maybe it is time to start look at it from the gameplay perspective? And add becuse VROM, VROM?

**Comment by davidpwbrown** [comment]
From the perspective of this CBM being a fuel cell, it is totally fine and realistic, if this DRAFT PR does not yet describe the CBM as a fuel cell, then it should, and probably will.
The internal furnace was silly and unrealistic, it burned anything you put into it, somehow, magically, inside your stomach.
A fuel cell can be small, light and efficient, hence why they are used extensively on space missions, These are two different things.
Realism is one of the design goals of this project, that is not going to change.
Ranting about " you should stop removing things for realism" and " you should do x instead, or do y and z " is not relevant to this PR. Please stop grinding old axes.

**Comment by kevingranade** [comment]
@Firestorm01X2 this is totally unreasonable behaviour. If you can't stay civil and refrain from derailing discussions, you are not going to be welcome to post here any more, this is your last warning.

As I pointed out in the first reply, a *fuel cell* would be a way to use gasoline to generate power, and I expect this PR to be adjusted to reflect that, but it just hasn't happened yet. If you want an ICE to generate power and use cable charging, absolutely that's a reasonable thing to do, and you can do it already.

As for your concerns about "poisoning",. it's being used in a technical sense meaning, "a buildup of chemicals that brings the intended reaction to a halt".  It has nothing to do with the produced chemicals being harmful to a *person*.

**Comment by jeremyshannon** [comment]
I'm pretty sure some of those compounds are actually toxic and would have to be flushed somewhere other than into your bloodstream, but I don't know what gave him the idea they'd just get dumped into your body. Even the Cata scientists wouldn't do that, because there's no reason to. Flushing would be a minor maintenance thing that should get abstracted into reloading the CBM.

 I believe the IRL problem with this is that these byproducts are hard to remove because they stick to things, and gum up the works of the cell. Inventing some materials that repel them, but still function for power generation could solve it, but you'd probably need some crazy sci-fi lab cranking out weird metamaterials.

**Comment by Firestorm01X2** [comment]
> @Firestorm01X2 this is totally unreasonable behaviour. If you can't stay civil and refrain from derailing discussions, you are not going to be welcome to post here any more, this is your last warning.

Wait a minute. Discussing things is unreasonable behaviour? 
I am not civil? Because I don't agree? What?

**Comment by davidpwbrown** [comment]
> > @Firestorm01X2 this is totally unreasonable behaviour. If you can't stay civil and refrain from derailing discussions, you are not going to be welcome to post here any more, this is your last warning.
> 
> Wait a minute. Discussing things is unreasonable behaviour?
> I am not civil? Because I don't agree? What?
> 
> > As for your concerns about "poisoning",. it's being used in a technical sense meaning, "a buildup of chemicals that brings the intended reaction to a halt". It has nothing to do with the produced chemicals being harmful to a person.
> 
> This I can understand.
> 
> But sentence about final warning are strange at best.

Did you miss the bit about derailing discussions?

**Comment by Firestorm01X2** [comment]
> Did you miss the bit about derailing discussions?

You are missing big corners of my point.
I am in fact think that this CBM are good an we need more of that. But there is problem.

Game design itslf shoud have some pricipals and stick to it.
Idea iteslf of that generator is great, from gameplay stand point. And that is it.

At the same time "Realism" itstlef is pretty abstract thing (Can someone at least formulate it?). 

You can put it on everything if you want and do not put if you don't want. You just need to find reason. Then why is there is the principles at all?

I am focussing you attention  to the fact that people trying really hard to find reason to add thing to the game from non gampelay stand point trying  really hard to attach it to abstract "Realism". 

To that exact "Realism" that was used to remove previous CBMs by similar principles.

In short
I am ranting not to:

>" you should stop removing things for realism" and " you should do x instead, or do y and z " is not relevant to this PR. Please stop grinding old axes.

But to that realism is abstract as hell and can be used for anything if you want. It is called chaos.

**Comment by davidpwbrown** [comment]
> > Did you miss the bit about derailing discussions?
> 
> You are missing big corners of my point.
> I am in fact think that this CBM are good an we need more of that. But there is problem.
> 
> Game design itslf shoud have some pricipals and stick to it.
> Idea iteslf of that generator is great, from gameplay stand point. And that is it.
> 
> At the same time "Realism" itstlef is pretty abstract thing (Can someone at least formulate it?).
> 
> You can put it on everything if you want and do not put if you don't want. You just need to find reason. And you will find it - we have sci-fi here. Then why is there is the principles at all?
> 
> I am focussing you attention to the fact that people trying really hard to find reason to add thing to the game from non gampelay stand point trying really hard to attach it to abstract "Realism".
> 
> To that exact "Realism" that was used to remove previous CBMs by similar principles.
> 
> In short
> I am ranting not to:
> 
> > " you should stop removing things for realism" and " you should do x instead, or do y and z " is not relevant to this PR. Please stop grinding old axes.
> 
> But to that realism is abstract as hell and can be used for anything if you want (especially in sci-fi setting). It is called chaos.

Post on the forums or subreddit to discuss overall project design goals and implementation.
Not on this PR.
With some adjustment, this PR should be fine and fit the projects goal of *striving* for realism, 
That has nothing to do with other things that were removed.
There is a clear distinction between the things that were removed previously, and this PR.
Realism is not abstract, realism is based on reality, therefore there is a pretty good guidebook for what can be included and what cannot.
You are not exactly getting the point here, to make it blunter, stop ranting about things about the projects overall design goals. 
This comments page is to offer feedback and reviews on this particular PR, given the starting point that the games design goals will *not* change, your comments are wildly unhelpful, off-topic and are derailing. 

If you do not stop, I suspect Kevin will ban you, I should probably not even engage with you, but I am hoping, somewhere, you will see sense, and stop.

**Comment by chaisawlajatang** [comment]
I tend to agree with @Firestorm01X2.
"Realism" is an incredibly abstract term, primo.
Internal combustion engine, implanted into abdomen is... really weird, secundo. It was mentioned above, I'll just summarize - look at it from the customer's point of view. Someone implants a battery, some energy-consuming implant (say, flashlight into one's forehead or foldable screwdrivers in place of fingers) and instead of plugging to electrical socket or charge from exernal accumulator the customer... gets a flask of highly flammable diesel and pours it into himself from sort of... funnel? and starts the engine? Heck, it even **sounds** as unbelievably unsafe thing. 

Cata's design, as it was stated several times, tries to achieve realism. But an _internal combustion engine just under one's liver_ is realistic in no way.

**Comment by davidpwbrown** [comment]
There is a difference between realistic and a good idea.
We are not talking like an engine in your abdomen chugging away like a tractor, we are talking about a fuel cell, something well established as being useful in various systems.
Now working from the premise that CBMS exist in the world of cataclysm, the next logical step is , how are they powered?

A fuel cell implanted into the body , would be one potential way of doing it, the science is understood, the impracticalities of it are potentially surmountable, it would be a bad idea for most people, for limited benefit, but not perhaps, for the military or those rich enough and with enough of a need. This then becomes a matter of rarity, not implementation.

A fuel cell could be very small, could run off of small amounts of chemicals for a long period of time, and only need refilling via recessed apertures every now and then, depending on their efficiency and output, for this purpose we would probably assume - low output, high efficiency. 

This is realistic, in terms of what we know is possible with current or near future technology, given the surrounding context in cataclysm of CBMs in general.

Yes it's weird, but it's not unrealistic, things like the internal furnace was unrealistic, because there was no known device or method of the small size required, that could just burn anything you swallowed, the whole idea was absurd and had no grounding in science.

This idea, however, is grounded in science, and yes could be absurd, but just because something would be impractical and absurd for the average consumer, does not mean it could and would not exist in the hands of those who would be willing.

So we can accept that in the world of cata, perhaps prototype special forces, behind enemy lines, CBM-enhanced soldiers existed, how would they charge their CBMS?

Solar panel technology in cata world isnt *vastly* better than ours, So they cant have solar recharging, theyd need huge area of panels to make it worhtwhile, that cannot really be folded into the body.

They cannot charge via cable, because they may not be back at base or in friendly territory very often.

They cannot carry all the batteries they need, for weight reasons.

The Joint Torsion CBM is a trickle charge harvesting leftover power from movement, not a lot of it, and it impedes movement when toggled active, not very useful in that scenario.

So how to keep their CBMS powered?  a small fuel cell implanted into them, that can keep them going for a few days perhaps, not something for every situation, and not something perhaps the general public would have, but feasible and suited for this purpose.

Theres a reason they were used on manned space missions, where things cannot just be refuelled/recharged easily.

I hope this answers the concerns pertaining to this particular PRs implementation.

I hope this clarifies things, and I hope no further griping about the projects overall design direction and goals muddy the waters of this PR, there are other places for that, and they are used frequently for that.

**Comment by esotericist** [comment]
> Cata's design, as it was stated several times, tries to achieve realism. But an _internal combustion engine just under one's liver_ is realistic in no way.

image

@chaisawlajatang you are arguing against a thing that has already been deemed not going to happen. If you're going to jump in on an argument please a) pay attention to the actual discussion, b) have something constructive to offer.

**Comment by Zireael07** [comment]
As-is, the description makes it on par with Internal Furnace as far as silliness goes. A diesel generator in abdominal cavity? WTF? Even as a military experiment, it's downright absurd, and it'd never pass any safety checks (unless the description is reworded to mention how exactly it's made safe, I assume it simply kills the user via energy discharge/reaction side effects)
I agree that the CBM is attractive from a gameplay point of view (a way to use up fuels for characters not using vehicles that much), but uh, can we get an explicit mention of fuel cells?

**Comment by Fris0uman** [comment]
Obviously the fuel generator is out, sorry I didn t have to update the draft to fuel cell yet.

**Comment by nexusmrsep** [comment]
A small enclosed module that contains a chemical reaction is not that
unbelievable if you can manage heat. I believe we all agree at this point
it's not a combustion engine. In that manner a lithium ion cell is a
similar concept. Put it in a body and you've got a power bank CBM. There
are conditions in which a diesel cell would not be valid, and it can be
learned by extrapolation of existing tech. Question is would be safely
reloadable, etc. But for now all we do here is orbit around ideas about
what this PR is not. My advice would be: name the science behind the
proposed CBM. Say clearly why it works, what are it's limitations etc. Then
we will be able to make proper discussion on the topic.

sob., 24 sie 2019, 16:34 użytkownik Zireael07 <notifications@github.com>
napisał:

**Comment by Edgar-cy-Lee** [comment]
Allow me to summarize the technology for Fris0uman...

Current CBM implementation is using Fuel Cell(Proper term is solid oxide fuel cell or SOFC) technology:
https://en.wikipedia.org/wiki/Solid_oxide_fuel_cell

> Advantages of this class of fuel cells include high combined heat and power efficiency, long-term stability, fuel flexibility, low emissions, and relatively low cost. The largest disadvantage is the high operating temperature which results in longer start-up times and mechanical and chemical compatibility issues.

Key issues with current technology was resolved with some high tech materials of CDDA:
1. High temperature requirement.(Please refer to LT-SOFC section. Lowering required temperature is possible and improves the efficiency)
2. Sulfur Filtering from fuel.
3. Method to clean up more viscous byproducts of reaction.
(other byproducts that does not stick to the cell can be cycled out)
4. Lifespan and durability.

**Comment by mlangsdorf** [comment]
BTW, here's the math:
1 Liter of gasoline: 1L at 34.1 MJ/L or 34.1 MJ (see data/json/items/fuels.json)
50% conversion: 17 MJ
1 battery charge is 1 kJ (see data/json/items/ammo.json)
1 L of gasoline converts to 17,000 battery charges.

I am pretty dubious about fitting a 50% efficient fuel cell and alternator into the human body.  As a reference, a 0.3L gasoline engine in CDDA is only 30% efficient in turning gasoline into kinetic energy, and the motorcycle bicycle alternators are only 48% efficient in turning kinetic energy into electricity.  So a small gasoline generator converts 1L of gasoline into 4,914 battery charges, but an even smaller engine running cooler inside the human body is 3x as efficient?  Seems off to me.

All the high efficiency fuel cell types tend to be high temperature designs, and while you can probably install a zinc-air fuel cell with a working temp below 40C in the human body, an alkaline fuel cell with a working temperature around 70C would probably be pretty unpleasant.

**Comment by kevingranade** [comment]
I'm 100% ok with saying, "it has to compromise on thermal envelope, size, and byproduct handling, so it doesn't hit anything close to peak performance for a non-CBM gasoline fuel cell".

Considering the MASSIVE gap between energy density of gasoline and most other sources, I suspect the only meaningful way to balance this is with conversion *rate* rather than efficiency.

We could crank efficiency all the way down to 10% (stretching the facts a great deal), and still have a cup of gasoline produce about a thousand charges, but it's totally reasonable to say the module is restricted in reaction rate by it's size, producing an arbitrarally small amount of power per unit of time.

In concrete terms, call it 25% efficient (about 8.5 J per mL) and as a stab in the dark, converts one mL per second while operating.

No joke, I didn't look back at the initial proposal until after writing that, so the similarity of the numbers is either coincidental or convergent calculations.

**Comment by mlangsdorf** [comment]
1 mL/s at 25% efficiency is 8 charges/second.  Definitely a top-tier CBM in that regard.

I don't know how easy it would be to do, but it would be reasonable to add a mild pain effect while the generator is in operation - the other 22 kJ/s that aren't being converted to electricity are going *somewhere*, and they're mostly going into heat being transferred through a heat sink that is necessarily somewhere close your body.  8 CBM charges/second at the cost of a little gasoline and +1 pain/second is an interesting trade-off, I think.

**Comment by Rail-Runner** [comment]
> 1 mL/s at 25% efficiency is 8 charges/second. Definitely a top-tier CBM in that regard.
> 
> I don't know how easy it would be to do, but it would be reasonable to add a mild pain effect while the generator is in operation - the other 22 kJ/s that aren't being converted to electricity are going _somewhere_, and they're mostly going into heat being transferred through a heat sink that is necessarily somewhere close your body. 8 CBM charges/second at the cost of a little gasoline and +1 pain/second is an interesting trade-off, I think.

If so, why not make it increase warmth by a lot rather than cause pain directly?

**Comment by mlangsdorf** [comment]
You're completely right, raising warmth is a better solution that I didn't think of immediately.  Then you can also use your generator CBM to help you explore ice labs =)

**Comment by chaisawlajatang** [comment]
Errr, 22kJ/s = 22 kilowatts, isn't it too much for an internal heater?

File: data/json/bionics.json [hunk 2]
```diff
@@ -1072,6 +1072,19 @@
     "occupied_bodyparts": [ [ "ARM_R", 1 ], [ "HAND_R", 2 ] ],
     "act_cost": 50
   },
+  {
+    "id": "bio_fuel_cell_gasoline",
+    "type": "bionic",
+    "name": "Gasoline Fuel Cell CBM",
+    "description": "A small gasoline fuel cell fixed to your scapula, despite its limited energy output compared to other fuel cell this device still produces significant amount of heat dissipated through heat exhaust protruding from your back.  A diffuse network of bio-plastic bladders has been meshed with your circulatory system and serves as a fuel tank.",
```
Comment (matskuman5) [inline data/json/bionics.json]:
```suggestion
    "description": "A small gasoline fuel cell fixed to your scapula. Despite its limited energy output compared to other fuel cells, this device still produces a significant amount of heat dissipated through a heat exhaust protruding from your back.  A diffuse network of bio-plastic bladders has been meshed with your circulatory system and serves as a fuel tank.",
```

**Comment by DaviBones** [comment]
If @chaisawlajatang's numbers are right, yes 22 kilowatts is a lot of heat. Assuming an average specific heat of 3.5 kJ/(kg*K), 22 kilowatts would heat up a 60 kg person up by 1 Kelvin about every 10 seconds, even if the heat was evenly distributed through the entire body.

**Comment by Moonsabrt** [comment]
> > We could crank efficiency all the way down to 10% (stretching the facts a great deal), and still have a cup of gasoline produce about a thousand charges, but it's totally reasonable to say the module is restricted in reaction rate by it's size, producing an arbitrarally small amount of power per unit of time.
> 
> A reserve power generator CBM makes a _lot_ of sense, especially for persons working far out of range of more conventional power resupply sources. Think about it this way: This is essentially the replacement for the minireactors. You jump over to the sub-prime with the rest of your collections team, and burn through all of your charges powering whatever. You can't just bring a generator with you because you have to stay mobile. So, instead of carrying a plutonium cell or whatever, you have about a cup of gasoline stored in you, that when you've run out of everything else, you can still supply power to your essential CBM's.
> 
> > You're completely right, raising warmth is a better solution that I didn't think of immediately. Then you can also use your generator CBM to help you explore ice labs =)
> 
> What about an optional cooling solution? If you have water available, you can use it to carry heat away from the CBM and your body. Or, if you choose to, you can pull water out of the body for the same purpose?

Don't those weird cooling jackets exist now a days?
Thats certainly a civilian option to handle heat, carry around a water jacket. Turn on your CBM and while your generating power refill/replace the water in  your jacket and let it cool you.

**Comment by SourSlime** [comment]
Did I read it wrong or has the activated cell phone (flashlight) been readded to the recycle center?

**Comment by kevingranade** [comment]
You're right, that's a merge error, will fix.
